src/Razor/test/Microsoft.VisualStudio.Razor.Integration.Test/InProcess/ErrorListInProcess.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.Razor.Integration.Test.InProcess
+{
+    internal class ErrorListInProcess : InProcComponent
+    {
+        public ErrorListInProcess(TestServices testServices)
+            : base(testServices)
+        {
+        }
+
+        public async Task<IEnumerable<IVsTaskItem>> GetErrorsAsync(CancellationToken cancellationToken)
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            var errorListItems = VsShellUtilities.GetErrorListItems(ServiceProvider.GlobalProvider);
```

Comment:
Talked with @aditya-kadam offline. `VsShellUtilities.GetErrorListItems` is an old API that we probably shouldn't use. The recommended alternative is `IDiagnosticViewerService`, but since we're not actually using this method yet I figured it was easier to just omit it than to try integrating with that. I'm sure we'll need this at some point and we can re-asses then.